### PR TITLE
feat: add federation_id endpoint

### PR DESCRIPTION
**Pull request by Kodylow**
We expect this in the guardian ui and I need it for guardian authentication there.

### CODE REVIEW

File: fedimint-core/src/config.rs [hunk 1]
```diff
@@ -120,6 +120,8 @@ pub struct PeerUrl {
 /// This includes global settings and client-side module configs.
 #[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize, Encodable, Decodable)]
 pub struct ClientConfig {
+    #[serde(default)]
```
Comment (dpc) [inline fedimint-core/src/config.rs:123]: We should add a better comment about requirements and constrains of evolving `ClientConfig` as from the top of my head, I don't remember them.

I think now that `ClientConfig` it's not signed a new field should be fine, and old clients will just ignore it, so LGTM?

File: fedimint-core/src/config.rs [hunk 2]
```diff
@@ -120,6 +120,8 @@ pub struct PeerUrl {
...
+    pub federation_id: Option<FederationId>,
```
Comment (m1sterc001guy) [inline fedimint-core/src/config.rs:124]: Why `Option`? To make it backwards compatible?

**Review by m1sterc001guy — DISMISSED** [review]

File: fedimint-core/src/config.rs (hunk 2 above)
Comment (Kodylow) [reply fedimint-core/src/config.rs:124]: Yes, happy to change and add versioning but this seemed more straightforward

Comment (dpc) [reply fedimint-core/src/config.rs:124]: Add a `// support(v0.2): can be changed into non-option` so we don't forget

**Review by dpc — APPROVED** [review]

**Review by elsirion — COMMENTED** [review]
Can we just add an API endpoint that returns the federation id instead? Adding redundant information in the client config sounds like a bad idea, especially when taking into account backwards/forwards compatibility.

**Comment by Kodylow** [comment]
Just changed to be federation_id api endpoint per eric, renaming pr

**Pull request closed without merging**